QUESTIONS.md

```diff
@@ -0,0 +1,297 @@
+# JS Fundamentals Questions
+
+**Do you know how and when o use let and/or const?**
+
+You use *let* for variables that are gonna change/reassign another value and const when you have a constant value in the variable
+
+*Example with 'const'*
+
+```
+const name = "Daniela"
+
+console.log(name)
+```
+
+*Example with 'let'*
+
+```
+for(let i = 0; i < n; i++) {
+    console.log(i);
+}
+```
+
+**How does the variables scope works?**
+
+Depending how you use the variables in your code is how you have accessibility and visibility of these ones. In other words, is when you can or cannot use/access to a variable, depending if this is in a block or not
+
+*Example using 'let':*
+
+```
+{ 
+    let number = 4;
+}
+
+// Number can't be used here
+```
+
+*Example using 'var':*
+
+```
+{
+    var number = 4
+}
+
+// Number can be used here
+
+```
+
+*Example with local scope:*
+
+```
+{
+    for(let i = 0; i < n; i++) {
+        const number = 10;
+    }
+}
+
+// Number cannot be used here
+```
+
+**When should I use *for each* vs *map*?**
+
+Map is used to transform each element of an array or either an object, it returns a new array 
+
+*Example using 'map':*
+
+```
+const array = [1, 2, 3, 4]
+
+array.map(function(e) {
+    return e * 2
+})
+
+console.log(array)
+
+El resultado será:
+[2, 4, 6, 8]
+
+```
+
+forEach is used to loop each element of an array or object 
+
+*Example using 'forEach':*
+
+```
+array.forEach(function(e) {
+    console.log(e * 2)
+})
+
+El resultado será:
+2
+4
+6
+8
+```
+
+**What is the *reduce* use for?**
+
+Is used to make calculations and it returns a value. The main idea of use reduce is to have a map and filter working together, you can filter the data that you have and map it in the same step
+
+*Example using 'reduce':*
+
+```
+const array = [1, 2, 3, 4]
+
+const initialValue = 10;
+
+const sumWithInitial = array.reduce((aux, current); => aux + current, initialValue
+);
+
+console.log(sumWithInitial)
+```
```

Comment:
Overall the answer is good, but an example with a filter, something like using `reduce` to only sum the odd numbers in an array would have been great.